From 8743194c17706a8818a2243f39e5e4c0ba647d76 Mon Sep 17 00:00:00 2001
From: gary-wzl77 <gary.wang@canonical.com>
Date: Mon, 20 Nov 2017 14:03:48 +0800
Subject: [PATCH] Link all dependencies in vendor folder to the custom go
 building environment.

---
 GNUmakefile | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/GNUmakefile b/GNUmakefile
index 1be8744..8a10534 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -118,5 +118,12 @@ static-assets:
 
 tools:
 	go get -u -v $(GOTOOLS)
+	@for dep in $(shell cd vendor; find * -type d -maxdepth 2 -mindepth 2); do \
+	echo "Linking $$dep into $$GOPATH"; \
+	rm -rf $$GOPATH/src/$$dep; \
+		mkdir -p $$GOPATH/src/$${dep%/*}; \
+	ln -fs $(shell pwd)/vendor/$$dep $$GOPATH/src/$$dep; \
+	done
+
 
 .PHONY: all ci bin dev dist cov test cover format vet ui static-assets tools vendorfmt
